git_perf/src/config.rs

```diff
@@ -245,149 +419,484 @@ epoch = "{}"
+            super::audit_dispersion_method("other_measurement"),
             git_perf_cli_types::DispersionMethod::StandardDeviation
         );
 
-        // Case 2: global setting
-        let configfile = r#"
+        // Test global setting
+        let global_config = r#"
 [audit.global]
 dispersion_method = "mad"
 "#;
+        fs::write(&local_config_path, global_config).unwrap();
         assert_eq!(
-            super::audit_dispersion_method_from_str(configfile, "any_measurement"),
+            super::audit_dispersion_method("any_measurement"),
             git_perf_cli_types::DispersionMethod::MedianAbsoluteDeviation
         );
 
-        // Case 3: precedence - measurement-specific overrides global
-        let configfile = r#"
+        // Test precedence - measurement-specific overrides global
+        let precedence_config = r#"
 [audit.global]
 dispersion_method = "mad"
 
 [audit.measurement."build_time"]
 dispersion_method = "stddev"
 "#;
+        fs::write(&local_config_path, precedence_config).unwrap();
         assert_eq!(
-            super::audit_dispersion_method_from_str(configfile, "build_time"),
+            super::audit_dispersion_method("build_time"),
             git_perf_cli_types::DispersionMethod::StandardDeviation
         );
         assert_eq!(
-            super::audit_dispersion_method_from_str(configfile, "other_measurement"),
+            super::audit_dispersion_method("other_measurement"),
             git_perf_cli_types::DispersionMethod::MedianAbsoluteDeviation
         );
 
-        // Case 4: no audit configuration (should return StandardDeviation)
-        let configfile = "";
+        // Test no config (should return StandardDeviation)
+        fs::remove_file(&local_config_path).unwrap();
         assert_eq!(
-            super::audit_dispersion_method_from_str(configfile, "any_measurement"),
+            super::audit_dispersion_method("any_measurement"),
             git_perf_cli_types::DispersionMethod::StandardDeviation
         );
+    }
+
+    #[test]
+    fn test_find_config_path_in_git_root() {
+        with_isolated_home(|temp_dir| {
+            // Create a git repository
+            env::set_current_dir(temp_dir).unwrap();
+
+            // Initialize git repository
+            std::process::Command::new("git")
+                .args(&["init", "--initial-branch=master"])
+                .output()
+                .expect("Failed to initialize git repository");
+
+            // Create config in git root
+            let config_path = temp_dir.join(".gitperfconfig");
+            fs::write(
+                &config_path,
+                "[measurement.\"test\"]\nepoch = \"12345678\"\n",
+            )
+            .unwrap();
+
+            // Test that find_config_path finds it
+            let found_path = find_config_path();
+            assert!(found_path.is_some());
+            assert_eq!(found_path.unwrap(), config_path);
```

Comment:
@cursoragent the git init calls are repetitive in the tests. Please refactor to have a single method to create a directory with an initialized git repository. Similarly the creation of HOME directories can be streamlined.